### PR TITLE
FMU / IO drivers: Move param definition of MAIN / AUX params to drivers

### PR DESCRIPTION
**Pull request by LorenzMeier**
This ensures param pruning works correctly on a per-driver level. This also adds the missing AUX params for FMUv5 which has 8 instead of previously 6 AUX ports.

@PX4/testflights Please test on Pixracer and Pixhawk 4.

### CODE REVIEW

File: src/drivers/px4io/px4io_params.c [hunk 1]
```diff
@@ -231,6 +231,126 @@ PARAM_DEFINE_FLOAT(PWM_MAIN_TRIM7, 0);
...
+ * When set to -1 the value for PWM_DISARMED will be used
+ *
+ * @reboot_required true
+ *
+ * @min -1
+ * @max 2200
+ * @unit us
+ * @group PWM Outputs
+ */
+PARAM_DEFINE_INT32(PWM_MAIN_DIS4, -1);
```
Comment (bkueng) [inline src/drivers/px4io/px4io_params.c]: These are also used for the fmu, so they need to be in the build, even if px4io is disabled.

**Comment by dagar** [comment]
These all need to go somewhere central (maybe something like a pwm output module interface) and then we need to do a pass to check usage across all modules.

The root cause of #9859 is that px4fmu is directly accessing PWM_AUX_REV and PWM_AUX_TRIM parameters https://github.com/PX4/Firmware/blob/master/src/drivers/px4fmu/fmu.cpp#L925-L942) even when it's configured as the primary pwm output (MAIN).

**Comment by dagar** [comment]
Followup https://github.com/PX4/Firmware/pull/9925

**Comment by santiago3dr** [comment]
flashed on a pixracer (V4)
ESC's do not initialize

**Comment by LorenzMeier** [comment]
@santiago3dr I presume you flashed this branch and not the follow-up PR from Daniel?

**Comment by santiago3dr** [comment]
correct, i flashed this branch, git hash: 023c905

**Comment by Avysuarez** [comment]
Flashed on a pixhawk 4 (V5)
https://review.px4.io/plot_app?log=ef0ce562-cb3a-4c8f-921d-3d4e4a852481
The ESC initialize and working well.

**Comment by santiago3dr** [comment]
tried again, same results, esc's do not initialize
here is a log from boot:
https://logs.px4.io/plot_app?log=b5e87ef0-ad99-42cb-a460-025e2fd44f7b

**Pull request closed without merging**